java/org/contikios/cooja/Cooja.java

```diff
@@ -2433,10 +2433,7 @@ public JDialog work() {
       public Simulation doInBackground() {
         Element root = new Element("simconf");
         if (cfgFile == null) {
-          var simElem = new Element("simulation");
-          simElem.addContent(getSimulation().getConfigXML());
-          root.addContent(simElem);
-          root.addContent(getPluginsConfigXML());
+          extractSimulationConfig(root);
```

Comment:
What do you think about doing something like this instead of adding an argument to `extractSimulationConfig()`?

`Element root = cfgFile != null ? extractSimulationConfig() : new Element("simconf");`